tests/paramgrill.c

```diff
@@ -1425,130 +1440,112 @@ static BMK_return_t BMK_benchMemInvertible(const buffers_t buf, const contexts_t
-            resultvar.cSize = tmpret.result.cSize;                                                              \
-            resultvar.cMem = tmpret.result.cMem;                                                                \
+        BMK_benchOutcome_t const outcome = BMK_benchMemInvertible(buf, ctx, BASE_CLEVEL, &cParams, mode, sec);  \
+        if (!BMK_isSuccessful_benchOutcome(outcome)) {                                                          \
+            DEBUGOUTPUT("Benchmarking failed\n");                                                               \
+            return ERROR_RESULT;                                                                                \
         }                                                                                                       \
-        if(mode != BMK_compressOnly) { resultvar.dSpeed = tmpret.result.dSpeed; }                               \
-    }                                                                                                           \
+        {   BMK_benchResult_t const tmpResult = BMK_extract_benchResult(outcome);                               \
+            if (mode != BMK_decodeOnly)  {                                                                      \
+                resultvar.cSpeed = tmpResult.cSpeed;                                                            \
+                resultvar.cSize = tmpResult.cSize;                                                              \
+                resultvar.cMem = tmpResult.cMem;                                                                \
+            }                                                                                                   \
+            if (mode != BMK_compressOnly) { resultvar.dSpeed = tmpResult.dSpeed; }                              \
+    }   }                                                                                                       \
 }
 
 /* Benchmarking which stops when we are sufficiently sure the solution is infeasible / worse than the winner */
-#define VARIANCE 1.2 
-static int allBench(BMK_result_t* resultPtr,
+#define VARIANCE 1.2
+static int allBench(BMK_benchResult_t* resultPtr,
                 const buffers_t buf, const contexts_t ctx,
                 const paramValues_t cParams,
                 const constraint_t target,
-                BMK_result_t* winnerResult, int feas) {
-    BMK_result_t resultMax, benchres;
+                BMK_benchResult_t* winnerResult, int feas)
+{
+    BMK_benchResult_t resultMax, benchres;
     U64 loopDurationC = 0, loopDurationD = 0;
     double uncertaintyConstantC = 3., uncertaintyConstantD = 3.;
     double winnerRS;
     /* initial benchmarking, gives exact ratio and memory, warms up future runs */
-    CBENCHMARK(1, benchres, tmp, BMK_both, BMK_iterMode, 1);
+    CBENCHMARK(1, benchres, tmp, BMK_both, 1);
```

Comment:
Did you find that this slows down paramgrill? I asked George if the complexity of iter mode was worthwhile, and he said it got a 3x? speedup.

Reply:
I did not checked `paramgrill` speed,
just verified that `paramgrill` was still running properly.

I suspect `BMK_iterMode` can be faster on small data sets,
since in this case it's possible to complete a compression / decompression test in a fraction of a second.
(I've been using `silesia.tar`, for which such conclusion is less true).

But a consequence of faster tests
would be an increased jitter in speed measurements, hence less reliable hill climbing.
That's why the initial code enforced a "1 second minimum per measurement" rule. If I do remember properly, it would even request a "2 second measurement period" (by default), meaning that, on average, 2 measurements will be taken, and the better one will be preserved. The goal of this policy is to reduce jitter to acceptable levels.

In George's code, jitter is taken into consideration by calculating `uncertaintyConstantX`, though it's limited to measured clock granularity. This is nonetheless much better than nothing.

Maybe we could use that capability to justify reducing measurement times ?

If that matters, I would suggest to keep using the `BMK_timeMode`, but make measurement time programmable (instead of the hard-coded "1 second per measurement"). We could then enforce a different (faster) policy within `paramgrill`.

As mentioned before, this is only going to be useful for short data sets,
and cost a bit of accuracy, hence potentially impact hill climbing.
The hill-climbing side effect is non-trivial to measure, and would require some dedicated tools.